wandb/interface/interface.py

```diff
@@ -214,14 +216,40 @@ def _summary_encode(self, value, path_from_root):
 
             return json_value
 
-    def _make_summary(self, summary_dict):
-        data = self._summary_encode(summary_dict, tuple())
-        summary = wandb_internal_pb2.SummaryRecord()
-        for k, v in six.iteritems(data):
-            update = summary.update.add()
-            update.key = k
-            update.value_json = json.dumps(json_friendly(v)[0], cls=WandBJSONEncoderOld)
-        return summary
+    def _make_summary(self, summary_record):
+        pb_summary_record = wandb_internal_pb2.SummaryRecord()
+
+        for item in summary_record.update:
+            pb_summary_item = pb_summary_record.update.add()
+            key_length = len(item.key)
+
+            assert key_length > 0
+
+            pb_summary_item.key = item.key[0]
+
+            if key_length > 1:
+                pb_summary_item.nested_key.extend(item.key[1:])
```

Comment:
lets use nested_key anytime we are updating nested keys (key_length > 1) and leave key unset in that case

and key is set only if key_length == 1
I think that definition makes more sense than using the top level key and nest_keys for later keys.